components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOImpl.java

```diff
@@ -1940,6 +1948,57 @@ public int getRoleAudienceRefId(String audience, String audienceId) throws Ident
         return id;
     }
 
+    @Override
+    public List<String> getMainRoleUUIDsForSharedRoles(List<String> sharedRoleUUIDs)
+            throws IdentityRoleManagementException {
+
+        List<String> mainRoleUUIDs = new ArrayList<>();
+        if (CollectionUtils.isEmpty(sharedRoleUUIDs)) {
+            return mainRoleUUIDs;
+        }
+
+        String query = GET_MAIN_ROLE_OF_A_SHARED_ROLE_SQL +
+                String.join(COMMA + WHITE_SPACE, Collections.nCopies(sharedRoleUUIDs.size(), QUESTION_MARK)) +
+                CLOSE_PARENTHESIS;
+
+        // To keep track of found sharedRoleUUIDs
+        Set<String> foundSharedRoleUUIDs = new HashSet<>();
+
+        try (Connection connection = IdentityDatabaseUtil.getUserDBConnection(false);
+             NamedPreparedStatement statement = new NamedPreparedStatement(connection, query)) {
+
+            for (int i = 0; i < sharedRoleUUIDs.size(); i++) {
+                statement.setString(i + 1, sharedRoleUUIDs.get(i));
+            }
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                while (resultSet.next()) {
+                    String mainRoleUUID = resultSet.getString(MAIN_ROLE_UUID);
+                    String sharedRoleUUID = resultSet.getString(SHARED_ROLE_UUID);
+
+                    // Add main role UUID to the result list
+                    mainRoleUUIDs.add(mainRoleUUID);
+
+                    // Mark the shared role UUID as found
+                    foundSharedRoleUUIDs.add(sharedRoleUUID);
+                }
+            }
+
+            // Add any sharedRoleUUID that was not found in the result as it represents a main role UUID
+            for (String sharedRoleUUID : sharedRoleUUIDs) {
+                if (!foundSharedRoleUUIDs.contains(sharedRoleUUID)) {
+                    mainRoleUUIDs.add(sharedRoleUUID);
+                }
+            }
+
```

Comment:
```suggestion
```

Reply:
Addressed and resolved in commit: [42705b9](https://github.com/wso2/carbon-identity-framework/pull/6162/commits/42705b984942a14ec734891444d31c9d33397a6f)